llama_toolchain/agentic_system/meta_reference/context_retriever.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
I think we should create a sub-directory called `rag` and move more of the logic from agent_instance to there while we are doing this